mmdet/structures/bbox/base_bbox.py

```diff
@@ -0,0 +1,412 @@
+        """Convert self to another box mode.
+
+        Args:
+            dst_mode (str or type): destination box mode.
+
+        Returns:
+            object: destination box mode object .
+        """
+        from .bbox_mode import convert_bbox_mode
+        return convert_bbox_mode(self, dst_mode=dst_mode)
+
+    @property
+    def bbox_dim(self) -> int:
+        """Return the value of ``self._bbox_dim``"""
+        return self._bbox_dim
+
+    def create_empty_bboxes(self: T,
+                            dtype: Optional[torch.dtype] = None,
+                            device: Optional[DeviceType] = None) -> T:
+        """Create a empty box with shape of (0, bbox_dim).
+
+        Args:
+            dtype (torch.dtype, Optional): data type of bboxes.
+            device (str or torch.device, Optional): device of bboxes.
+
+        Returns:
+            T: Empty box instance.
+        """
+        empty_bboxes = self.tensor.new_zeros((0, self._bbox_dim),
+                                             dtype=dtype,
+                                             device=device)
+        return type(self)(empty_bboxes)
+
+    def create_fake_bboxes(self: T,
+                           sizes: Tuple[int],
+                           fill: float = 0,
+                           dtype: Optional[torch.dtype] = None,
+                           device: Optional[DeviceType] = None) -> T:
+        """Create specific size fake boxes.
+
+        Args:
+            sizes (Tuple[int]): The size of fake boxes.
+            fill (float): filling value.
+            dtype (torch.dtype, Optional): data type of bboxes.
+            device (str or torch.device, Optional): device of bboxes.
+
+        Returns:
+            T: Empty box instance.
+        """
+        fake_bboxes = self.tensor.new_full(
+            sizes, fill, dtype=dtype, device=device)
+        return type(self)(fake_bboxes)
+
+    def __getitem__(self: T, index: IndexType) -> T:
+        """Rewrite getitem to protect the last dimension shape."""
+        bboxes = self.tensor
+        if isinstance(index, Tensor) and index.dtype == torch.bool:
+            assert index.dim() < bboxes.dim()
+        elif isinstance(index, tuple):
+            assert len(index) < bboxes.dim()
+            if Ellipsis in index:
+                assert index[-1] is Ellipsis
+
+        bboxes = bboxes[index]
+        if bboxes.dim() == 1:
+            bboxes = bboxes.reshape(1, -1)
+        return type(self)(bboxes)
+
+    def __setitem__(self: T, index: IndexType, values: Union[Tensor, T]) -> T:
+        """Rewrite setitem to protect the last dimension shape."""
+        assert type(values) is type(self)
+        values = values.tensor
+
+        if isinstance(index, Tensor) and index.dtype == torch.bool:
+            assert index.dim() < self.tensor.dim()
+        elif isinstance(index, tuple):
+            assert len(index) < self.tensor.dim()
+            if Ellipsis in index:
```

Comment:
When is Ellipsis used ？

Reply:
It enables `Boxes[None, ...]` and disables the `Boxes[..., None]`